airootfs/usr/share/vx-img/flash-image.sh

```diff
@@ -89,39 +126,71 @@ _hashash=0
 _toflash=""
 _images=()
 _extensions=()
+
+_matches=()
+
 for f in "$_path"/*; do
     _filename="${f##*/}"
     _extension="${_filename##*.}"
 
-    if [[ "$_extension" == "gz" || "$_extension" == "lz4" ]]; then
+    if (echo "$_filename" | grep -Po "$_match") ; then
+        _matches+=("$_filename")
+    elif [[ "$_extension" == "gz" || "$_extension" == "lz4" ]]; then
         _images+=("$_filename")
         _extensions+=("$_extension")
     elif [[ "$_extension" == "sha256sum" ]]; then
         _hashash=0
     fi
 done
 
-if [[ -z ${_images[0]} ]]; then
+if [[ -z ${_matches[0]} &&  -z ${_images[0]} ]]; then
     echo "Found no image to flash. Exiting..."
     exit
 fi
 
-echo "Found the following images."
-i=1
-for img in "${_images[@]}"; do
-    echo "$i. $img"
-    ((i+=1))
-done
-
-echo  "Please select one to flash [${_images[-1]}]"
-read -r answer
+if [[ "${#_matches[@]}" == 1 ]]; then
+    echo "Found only one image in the right format."
+    _toflash=${_matches[0]}
+    _extension="gz"
+    _finalsize=$(echo "$_toflash" | grep -oP $_sizematch)
+elif [[ -n ${_matches[0]} ]]; then
+    echo "Found several images that match the expected format."
+    unset answer
+    menu "${_matches[@]}" "Please select an image to flash" 
+
+    if [[ $err == 1 ]]; then
+        echo "Something went wrong, please try again."
+        exit
+    fi
 
-if [[ -n $answer ]]; then
-    _toflash=${_images[answer-1]}
-    _extension=${_extensions[answer-1]}
+    if [[ -n $answer ]]; then
+        _toflash=${_matches[$answer-1]}
+    else
+        _toflash=${_matches[-1]}
+    fi
+    _extension="gz"
+    _finalsize=$(echo "$_toflash" | grep -oP $_sizematch)
+elif [[ "${#_images[@]}" == 1 ]]; then
+    echo "Found only one image that might work."
+    _toflash=${_images[0]}
+    _extension=${_extensions[0]}
 else
-    _toflash=${_images[-1]}
-    _extension=${_extensions[-1]}
+    echo "Found the following images that might work."
```

Comment:
This is kind of an edge case but its a little weird to me that if there are both multiple images that match and multiple other files it doesn't give you all the options. If you move the "if only one image" elif up, I think we could combine the multiple matches and multiple images into one final else step that just gives all of either option.

Reply:
I can do that. My thinking was that we wanted to prioritize images with the right filename format, and look for nonstandard names as a last resort. If we don't want to do that then I can combine them.